riscv/src/runtime.rs

```diff
@@ -281,29 +281,69 @@ impl Runtime {
             "std::machines::hash::keccakf::KeccakF",
             None,
             "keccakf",
-            vec![],
-            [format!(
-                "instr keccakf link ~> keccakf.keccakf({}, {}, STEP);",
-                reg(0),
-                reg(1)
-            )],
-            2,
-            std::iter::once(format!("{} <=X= 0x100;", reg(0))) // filler value for input pointer
-                .chain(std::iter::once(format!("{} <=X= 0x300;", reg(1)))) // filler value for output pointer (at least 200 bytes away)
-                .chain(std::iter::once("keccakf;".to_string())) // must be called at least once
+            vec!["memory"],
+            [r#"instr keccakf X, Y
+                    link ~> tmp1_col = regs.mload(X, STEP),
+                    link ~> tmp2_col = regs.mload(Y, STEP + 1)
+                    link ~> keccakf.keccakf(tmp1_col, tmp2_col, STEP)
+                {
+                    // make sure tmp1_col and tmp2_col are aligned memory addresses
+                    tmp3_col * 4 = tmp1_col,
+                    tmp4_col * 4 = tmp2_col,
+                    // make sure the factors fit in 32 bits
+                    tmp3_col = X_b1 + X_b2 * 0x100 + X_b3 * 0x10000 + X_b4 * 0x1000000,
+                    tmp4_col = Y_b5 + Y_b6 * 0x100 + Y_b7 * 0x10000 + Y_b8 * 0x1000000
+                }
+            "#
+            .to_string()],
+            0,
+            std::iter::once("set_reg 10, 0x100;".to_string()) // filler value for input pointer
+                .chain(std::iter::once("set_reg 11, 0x300;".to_string())) // filler value for output pointer (at least 200 bytes away)
```

Comment:
what are these filler values actually?

Reply:
These are memory addresses for the registers used in the init call. They need to be at least 200 bytes apart as that's the length of the input (25 64-bit values) and the output (25 64-bit values).

Reply:
I think we could use the same address for input and output. Also, we need to change this for continuations, but fine for now, I'll approve.